src/interchange/src/avro/schema.rs

```diff
@@ -91,27 +91,33 @@ fn get_named_columns<'a>(
                 let name = if vs.len() == 1 || (vs.len() == 2 && vs.iter().any(is_null)) {
                     // There is only one non-null variant in the
                     // union, so we can use the field name directly.
-                    base_name.map(|n| n.to_owned()).or_else(|| {
-                        v.get_piece_and_name(schema.root)
-                            .1
-                            .map(|full_name| full_name.base_name().to_owned())
-                    })
+                    base_name
+                        .map(|n| n.to_owned())
+                        .or_else(|| {
+                            v.get_piece_and_name(schema.root)
+                                .1
+                                .map(|full_name| full_name.base_name().to_owned())
+                        })
+                        .unwrap_or_else(|| "?column?".into())
```

Comment:
I tried to do this change also and this line is where I got stuck. I'm not sure what the purpose of `validate_schema_1` is so it's a bit difficult to comment. But: should `?column?` be a const or something so it's more obvious when it is produced? Does it need to have the postgres semantics of what to call an unnamed column? Or is this a schema coming from avro and so we can name these columns whatever we like, it's just that we happened to choose the same name as the unnamed column?

Reply:
> Or is this a schema coming from avro and so we can name these columns whatever we like, it's just that we happened to choose the same name as the unnamed column?

Precisely this. I'm going to put together another PR after this one that does something smarter here. We should just name the column after the type or something. But it needs tests, plus is a potentially breaking change, so wanted to split it out into its own PR.

The upshot, though, is I expect that `?column?` will ultimately be a thing that gets created in just one or two places in the `sql` crate, so didn't want to go to the trouble of creating a `UNNAMED_COLUMN_NAME` const in some root crate just to delete it in a follow-up PR.

Reply:
Ok, I'm fine not making it a const if it's going to be removed soon.

What happens if there are multiple unnamed columns in an avro schema? Will they have multiple columns named `?column?` and won't be able to reference any of them due to ambiguity?

Reply:
Yeah exactly. It's "fine" because you can alias the source columns `SELECT * FROM dumb_source (goodname1, goodname2)` when you actually reference the source.